metron-stellar/stellar-common/src/test/java/org/apache/metron/stellar/dsl/functions/HashFunctionsTest.java

```diff
@@ -0,0 +1,169 @@
+
+  @Test(expected = IllegalArgumentException.class)
+  public void nullArgumentListShouldThrowException() throws Exception {
+    hash.apply(null);
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void emptyArgumentListShouldThrowException() throws Exception {
+    hash.apply(Collections.emptyList());
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void singleArgumentListShouldThrowException() throws Exception {
+    hash.apply(Collections.singletonList("some value."));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void argumentListWithMoreThanTwoValuesShouldThrowException3() throws Exception {
+    hash.apply(Arrays.asList("1", "2", "3"));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void argumentListWithMoreThanTwoValuesShouldThrowException4() throws Exception {
+    hash.apply(Arrays.asList("1", "2", "3", "4"));
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void invalidAlgorithmArgumentShouldThrowException() throws Exception {
+    hash.apply(Arrays.asList("value to hash", "invalidAlgorithm"));
+  }
+
+  @Test
+  public void invalidNullAlgorithmArgumentShouldThrowException() throws Exception {
+    assertNull(hash.apply(Arrays.asList("value to hash", null)));
+  }
+
+  @Test
+  public void nullInputForValueToHashShouldProperlyThrowException() throws Exception {
+    assertNull(hash.apply(Arrays.asList(null, "md5")));
+  }
+
+  @Test
+  public void allAlgorithmsForMessageDigestShouldBeAbleToHash() throws Exception {
+    final String valueToHash = "My value to hash";
+    final Set<String> algorithms = Security.getAlgorithms("MessageDigest");
+
+    algorithms.forEach(algorithm -> {
+      try {
+        final MessageDigest m = MessageDigest.getInstance(algorithm);
+        m.update(valueToHash.getBytes(StandardCharsets.UTF_8));
+
+        assertEquals(BaseEncoding.base16().encode(m.digest()), hash.apply(Arrays.asList(valueToHash, algorithm)));
+      } catch (NoSuchAlgorithmException e) {
+        throw new RuntimeException(e);
+      }
+    });
+  }
+
+  @Test
+  public void allAlgorithmsForMessageDigestShouldBeAbleToHashDirectStellarCall() throws Exception {
+    final String valueToHash = "My value to hash";
+    final Set<String> algorithms = Security.getAlgorithms("MessageDigest");
+
+    algorithms.forEach(algorithm -> {
+      try {
+        final Object actual = run("HASH('" + valueToHash + "', '" + algorithm + "')", Collections.emptyMap());
+
+        final MessageDigest m = MessageDigest.getInstance(algorithm);
+        m.update(valueToHash.getBytes(StandardCharsets.UTF_8));
+
+        assertEquals(BaseEncoding.base16().encode(m.digest()), actual);
+      } catch (NoSuchAlgorithmException e) {
+        throw new RuntimeException(e);
+      }
+    });
+  }
+
+  @Test
+  public void nonStringValueThatIsSerializableShouldBeSerializedAndThenEncodedUsingHex() throws Exception {
```

Comment:
As noted, it's a nit.  Do as you choose, it won't interfere with a +1 from me, after the substantive items are taken care of.  My inclination, though, is that verbiage (beyond what's needed for clarity) should go in comments.